### ISSUE TITLE
Duplicate tab as child tab does not respect child tab insertion position setting

### ISSUE THREAD

**Issue by lbmaian** [piroor/treestyletab#1478]
## Short description
Duplicate tab as child tab does not respect child tab insertion position setting.

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Set TST duplicate tab setting to "a child of the current tab"
 4. Set TST child tab insertion position setting to "Append to the end of the tree"
 5. Tree structure:
```
A
\- B
```
 6. Duplicate tab A (use TST fake context menu's duplicate tab until #1477 is fixed)

## Expected result
```
A
|- B
\- A_dup
```

## Actual result
```
A
|- A_dup
\- B
```

## Environment

 * Platform (OS): Windows 10 Pro, Version 1709, Build 16299.19
 * Version of Firefox: Nightly 58.0a1 (2017-10-18) (64-bit)
 * Version (or revision) of Tree Style Tab: 2.0.7

**Comment by piroor** [comment]
After I researched I've realized that this is caused by attachTabTo and behaveAutoAttachedTab. When the tab is duplicated, the tab is automatically moved next to the original tab at first (by TST), then behaveAutoAttachedTab is called. It calls attachTabTo internally, however, because the duplicated tab is attached to the original tab, attachTabTo does nothing. As the result, the duplicated tab is not processed by attachTabTo and not moved to the end of existing children.
